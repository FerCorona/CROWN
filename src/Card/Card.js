import React from 'react';

import Button from './../Button';

const Card = ({ src, title, content, buttonConfig, customButton }) => (
  <div className='CROWNCard'>
    <div
      className='Card'
      style={
        {
          backgroundImage: `url(${src})`,
          backgroundPosition: 'center',
          backgroundRepeat: 'no-repeat',
          backgroundSize: 'cover',
          position: 'relative'
        }
      }>
      <div className='Content'>
        <div className='Title'>{title}</div>
        <div className='ContentCard'>{content}</div>
        <div className='Others'>{customButton ? customButton : <Button {...buttonConfig} />}</div>
      </div>
    </div>
  </div>
);

Card.defaultProps = {
  src: 'linear-gradient(var(--red-4), var(--pink-1))',
  title: 'Card Title',
  content: 'Card Content  ---- Card Content ---- Card Content  ---- Card Content ---- Card Content  ---- Card Content ---- Card Content  ---- Card Content ----',
  buttonConfig: {
    shape: 'Square',
    color: '--gray-blue-3',
    text: 'Aceptar',
    onClick: () => { }
  },
  customButton: false
};

export default Card;