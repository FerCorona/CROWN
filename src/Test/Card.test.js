import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../Card/index';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Card />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});