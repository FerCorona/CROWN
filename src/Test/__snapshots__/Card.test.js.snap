// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="CROWNCard "
>
  <div
    className="Card"
    style={
      Object {
        "backgroundImage": "url(linear-gradient(var(--red-4), var(--pink-1)))",
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "position": "relative",
      }
    }
  >
    <div
      className="Content"
    >
      <div
        className="Title"
      >
        Card Title
      </div>
      <div
        className="ContentCard"
      >
        Card Content  ---- Card Content ---- Card Content  ---- Card Content ---- Card Content  ---- Card Content ---- Card Content  ---- Card Content ----
      </div>
      <div
        className="Others"
      >
        <a
          className="CROWNButton Square "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "var(--gray-blue-3)",
              "boxShadow": "0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)",
              "color": "var(--white)",
            }
          }
        >
          Aceptar
        </a>
      </div>
    </div>
  </div>
</div>
`;
